### ISSUE TITLE
"Return to merchant" Show this message.

### ISSUE THREAD

**Issue by Jiks19** [braintree/braintree-android-drop-in#53]
![screenshot_2017-11-13-16-13-11-528_com android chrome](https://user-images.githubusercontent.com/22051913/32725250-56fe1878-c89a-11e7-8fee-55fd446b64d0.png)

I have use braintree-android-drop-in.
I try with my SendBox testing credentials

Not found any error or issue.
How to resolve it?

Thanks,

**Comment by sdcoffey** [comment]
Hey @Jiks19,

Thanks for submitting this bug report. Can you be a little more specific about what precisely is wrong here?

**Comment by Jiks19** [comment]
Hi sdcoffey,

Thanks for replay.

![01](https://user-images.githubusercontent.com/22051913/32818702-d24195cc-c9ea-11e7-9988-0e404f0dfd89.PNG)

I have use yellow mark credential.
1) Click on Paypal button.
2) Enter login/password.
3) Click on login button.

than return "Return to merchant" screen.

4) Click on return to merchant button than return in my app.
not find any error or other exception.


https://sandbox.braintreegateway.com/
Go to
Settings > Processing > payment method > PayPal

1) Link Sendbox
2) Entered PayPal email, client id and client secret
3) Click on Link PayPal Sendbox.

I have share all the details.
Let me inform me any more detail you need.

Thanks.

**Comment by sdcoffey** [comment]
He @Jiks19,

Thanks for the additional information. Unfortunately, we were not able to reproduce the issue using the steps you provided for us. Can you share with us the URLs that open in the browser when the switch starts? We can use that information to debug what might be happening in this particular case.

**Comment by Jiks19** [comment]
Hi sdcoffey,

can you provide any testing credentials(email/password) for India country?
So i can test here with your testing credentials. 

Thanks,

**Comment by Jiks19** [comment]
This URL open in browser

https://www.sandbox.paypal.com/agreements/approve?nolegacy=1&ba_token=BA-31N214971P224670Y&useraction=#/checkout/genericError?code=VU5TVVBQT1JURURfUEFZRUVfQ1VSUkVOQ1k%3D

**Comment by zakirhusain92** [comment]
I am facing same problem with woocommerce custom payment gatweway for paypal. Please let me know If you find any solution

**Comment by sivakumarcse12** [comment]
I am also facing same problem with simple core php in both mode live and sandbox, How to fix any one give suggestion... Thanks in advance

**Comment by Vaibs** [comment]
I am facing the same issue. I followed a paypal php example , https://www.evoluted.net/thinktank/web-development/paypal-php-integration

I create 2 sandbox account, one business and the other buyer.

Now after setting up the application, while testing the buyers amount should be deducted with the sellers account should be credited. Whats happening is that weird page on the top with the button saying return to merchant.

Please help

**Comment by ArjunDhar** [comment]
I'm facing the same issue. And everything was working fine earlier. Same issue in **Sandbox** and **production**. This page is accompanied by a URL [https://www.sandbox.paypal.com/checkoutnow?nolegacy=1&token=EC-459099459E9828727#/checkout/genericError?code=VU5TVVBQT1JURURfUEFZRUVfQ1VSUkVOQ1k%3D](url)

also in the Merchant Account > Dev console @ [https://developer.paypal.com/developer/dashboard/live/](url) ; there is no error. 

So basically there is no error on the front end and no error in their backend. Smells like a Bug or bad error handling by BrainTree.

**Comment by ArjunDhar** [comment]
Hi an update to my post above. So in the response JSON, i took out some clues:
`"data":{"memberSegment":"SERVICE_ERROR"} .... "contingency":"UNSUPPORTED_PAYEE_CURRENCY" (in JSON Response)`
However, I have verified many times I am sending USD. So not sure why it insists `UNSUPPORTED_PAYEE_CURRENCY`

**Comment by ArjunDhar** [comment]
@sdcoffey and others : I jut looked @ the names of all the posters. Their names seem "**Indian**". Pay Pal recently launched in India and based on an error I managed to grab on, it said "**Transactions between Indian residents must be made using Indian Rupees**" . lol

So its assuming Calls made from India (perhaps based on IP) are intended for Indian Merchants. But Our PayPal is for US customers visiting the site. So I think there is a bug @ PayPal or atleast this is not properly handled. 

Either you need to communicate to your merchants some regulatory requirement they are missing in their ADMIN or you have an out right bug in handling this properly. 

I think this bug is very specific to India. Please followup or have someone followup.

**Comment by Vaibs** [comment]
This should not be the feature cause(as a seller) my buyers can reside anywhere in the world and using their own countries pp account . their might be something else.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by ArjunDhar** [comment]
Yes Exactly. Thats what im saying. They have some regulatory RULE in their that isnt smart enough or some regulatory requirement for the  merchant is missing, Since they just officially launched in India its possible they have setup extra rules that are interfering in normal processing.

Based on all the data I have gathered while debugging:
1. Its surely a Currency related issue -- UNSUPPORTED_PAYEE_CURRENCY
2. It said Transactions between Indian residents must be made using Indian Rupees
3. And all the guys on this thread are complaining seem to be from India + Paypal launched less than a month back and this thread is less than that old

...do the maths. The cause is obvious  They need to fix it or tell us how to fix it.

**Comment by abhilashpullelil** [comment]
Is there any update on this issue? Can't Indian merchant's accept USD from Indian customers? If no then how can an Indian merchant accepting money from Indian customer in PayPal?

**Comment by prateekkatyal** [comment]
Same issue here. 

I contacted their Merchant Support and they have confirmed that Indian currency transactions are not yet live. They said that it'll go live in February/March

Right now it seems I can't even accept USD payments from Indian residents?

Has anybody found a solution?

**Comment by Vaibs** [comment]
Share me your sample code, might be i cam help. I am not sure why it's working for me and even i went out through the same problem.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by dineshbhosale** [comment]
This is happening on my end as well

**Comment by bluk** [comment]
Hey all,

As a quick update, since it seems like many of you are experiencing the same issue: @prateekkatyal is correct in saying that this is an issue _specifically_ with India-based PayPal merchant accounts. India-based PayPal merchant accounts should only use single payment checkout flows at this time. The [Express Checkout with Braintree](https://developer.paypal.com/docs/accept-payments/express-checkout/ec-braintree-sdk/get-started/) solution is the officially supported integration flow.

If you want to go live, please contact [Merchant Technical Support](https://www.paypal-techsupport.com/) to get integration assistance and to ensure your account meets all of the requirements.

**Comment by dineshbhosale** [comment]
I wasted like 7 months to get this to work :( 

Because I'm a solo developer I thought there must be some error on my end.
So I ended up changing implementation of Paypal SDK more than 3 times in my golang project.

After trying hard enough, I got this to work by using a buy now button with IPN listener by using some hacks such as passing user ID in custom field etc.

I got ill and was ICU for few weeks to get this to work in less time.

I followed suggestions that were given in `github.com/logpacker/PayPal-Go-SDK`, It worked fine in dev mode but as soon as I switched to production mode it didn't work.

As a result I had implement one more payment option (Instamojo) which took more time to get the actual product ready. 

I had no issues with Instamojo even though they do not provide a package for golang, they even provide option to specify webhook URL for payment notifications.

I feel so bad right now, I don't even know what suggestion to give to make this to work without any issues for Indian merchants, but still here is a list that I have made that may ease the process.

- Please make SDKs fail in dev mode itself so we will know that it will not work in production.
- Add more golang examples and a sample application that will work in dev and production mode for Indian merchants without any examples.

**Comment by dineshbhosale** [comment]
I'm having same issues with `github.com/logpacker/PayPal-Go-SDK`,

```
c.CreateDirectPaypalPayment
```

Works and creates a redirect link.

After redirect link is opened it shows "Return to merchant" , sometimes shows "Unsupported payee currency" in production mode.

I wasted like 7 months to get this to work :( 

Because I'm a solo developer I thought there must be some error on my end.
So I ended up changing implementation of Paypal SDK more than 3 times in my golang project.

After trying hard enough, I got this to work by using a buy now button with IPN listener by using some hacks such as passing user ID in custom field etc.

I got ill and was ICU for few weeks to get this to work in less time.

I followed suggestions that were given in `github.com/logpacker/PayPal-Go-SDK`, It worked fine in dev mode but as soon as I switched to production mode it didn't work.

As a result I had implement one more payment option (Instamojo) which took more time to get the actual product ready. 

I had no issues with Instamojo even though they do not provide a package for golang, they even provide option to specify webhook URL for payment notifications.

I feel so bad right now, I don't even know what suggestion to give to make this to work without any issues for Indian merchants, but still here is a list that I have made that may ease the process.

**Comment by bluk** [comment]
@dineshbhosale I apologize for the difficulties you've had in your integration. I suggest contacting [PayPal Merchant Technical Support](https://www.paypal-techsupport.com/) as they have the best resources to ensure that any India-based merchant integration can be successful.

**Comment by choudharysuman** [comment]
I am getting same problem when i got to live app. On sandbox it is working fine. But on live getting below url:
https://www.paypal.com/cgi-bin/webscr?cmd=_express-checkout&token=EC-32A91625D19611252#/checkout/genericError?code=VU5TVVBQT1JURURfUEFZRUVfQ1VSUkVOQ1k%3D

Anyone please assist to me.
